### PR TITLE
Update sci-mathematics/amath to 1.8.5

### PR DESCRIPTION
**Pull request by rainlance**
Changes since 1.8.3
- Fix build on compilers with broken endian detection.
- Ignore casing in hexadecimal numbers.

### CODE REVIEW

File: sci-mathematics/amath/amath-1.8.5.recipe [hunk 1]
```diff
@@ -44,5 +44,5 @@ INSTALL()
 
 TEST()
 {
-	make test
+	make --test
```
Comment (alaviss) [inline sci-mathematics/amath/amath-1.8.5.recipe]: There's no `--test` switch for make.

Comment (rainlance) [reply sci-mathematics/amath/amath-1.8.5.recipe]: The switch is misplaced. It was intended as a switch for the application itself and not for the make file.
I'm am not so familiar with Github pull request. Should I make a new pull request in order to remove changes in line 47?

Comment (fbrosson) [reply sci-mathematics/amath/amath-1.8.5.recipe]: Thanks @rainlance for your PR!

> Should I make a new pull request in order to remove changes in line 47?

You don't need to make a new pull request. You can amend your commit with `git commit --amend amath-1.8.5.recipe` and then "force push" it with `git push -f`.

BTW, I just noticed that your commit is in your master branch.
While this is not a problem at all, many users prefer to use [feature-branches](https://github.com/haikuports/haikuports/wiki/DevelopmentModel#what-are-feature-branches).

https://github.com/haikuports/haikuports/wiki/DevelopmentModel#what-are-feature-branches
This allows you to keep your master branch perfectly in sync with the project's master, and even easily refresh your master branch at any time, without having to touch your feature-branches.
That said, you are perfectly free to submit anything from your master branch. Using feature-branches is just a recommendation, and not using them does not have any impact.

(It's fine to continue with this PR. Actually, it is perfectly OK, and even (slightly) better than closing it and opening a new PR, but if you prefer to do that, it's also OK.)

If you wish I can "fix" your commit in this PR for you. (I'm used to updating PRs, so it's an easy task for me.).
BTW, it won't have any impact on the final result and you will still be the author of the commit.
Shall I do so?

Comment (fbrosson) [reply sci-mathematics/amath/amath-1.8.5.recipe]:
```diff
 TEST()
 {
+	LIBRARY_PATH="$sourceDir/src/lib:$sourceDir/src/cplex:$sourceDir/src/real:$sourceDir/src/clib${LIBRARY_PATH:+:$LIBRARY_PATH}" \
 	make test
}
```
BTW, amath builds fine on x86_64 (**without** the `--without-stdc++`) and passes most tests on this platform:
```
Passed: 541, failed: 1
```
The only test which fails is:
```
FAIL: [coth(1+2i)] expected [coth(1+2i) = 0.82132979749385+0.17138361290919i] but got [coth(1+2i) = 0.82132979749385+0.17138361290918i]
```
The difference is in the very last digit of the imaginary part:

Expected | 0.17138361290919
-- | --
Got | 0.17138361290918

Comment (rainlance) [reply sci-mathematics/amath/amath-1.8.5.recipe]: And thank you for the thoroughly review !

**Comment by fbrosson** [comment]
Sorry @rainlance for having taken so long to push my little fix, and thanks for your interest in Haiku and for being the author of amath!

So, here is my update. What do you think of the commit message? Shall I use it for the squashed merge (in place of your commit message)? (The squashed commit will retain your authorship.)

BTW, please do not hesitate to suggest any changes in the commit message or even in the recipe itself.

**Comment by fbrosson** [comment]
I did not try to find out why the tests are crashing on x86_gcc2. All I can say right now is that on that arch the tests crash after:
```
PASS: [cbrt(120+75i) = 5.1210176499+0.96470708458i]
PASS: [cbrt(120-75i) = 5.1210176499-0.96470708458i]
```
while on the other architectures the tests continue with the next tests:
```
PASS: [ln(20+40i) = 3.8004512298+1.1071487178i]
PASS: [ln(20-40i) = 3.8004512298-1.1071487178i]
...
```
So I thought `amath "ln(20+40i)"` would crash on x86_gcc2, but it did not. Actually, it gives a correct answer :)

BTW, that's why I finally dropped a change I was planning to include in my update:
```diff
-ARCHITECTURES="x86_gcc2 x86 x86_64"
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
```
BTW, I'd say it is OK to mark `x86_gcc2` as tested, but if you prefer we can mark it as untested, i.e. keep the above change in `ARCHITECTURES`.
If we mark amath as untested on x86_gcc2 then the only package which will be available (from Haiku Depot or with pkgman) on x86_gcc2 will be amath_x86, and it would only be available for users of x86_gcc2 hybrid.
That said, users of x86_gcc2 would still be able to build amath.

Anyway, right now amath is marked as tested on all architectures and I guess this is OK.
Shall I _**squash** and merge_ your PR?

**Comment by fbrosson** [comment]
Good news: replacing `make test` by `amath --test` was enough to make amath pass all tests on x86_gcc2.
So we don't need to add support for x86 secondary arch anymore.
(Well, we still can, if we want to have it built with a recent version of gcc.)

**Comment by rainlance** [comment]
@fbrosson No worries. Your changes are more than what I would be able to achieve! I am still learning the basics of Haiku OS. The Haiku API is really nice though with all the builtin GUI support. Git is also still new to me. Used to be on private TFS.

**Comment by fbrosson** [comment]
> I am still learning the basics of Haiku ~OS~.

Cool!

BTW, the official name (of the OS) is just "Haiku". (The haiku dot org domain was already taken and it seems its owner was not willing to give it for a reasonable price... so haiku-os.org was chosen, but the official name of the OS remains Haiku.)

**Pull request merged**